controllers/object_controls.go

```diff
@@ -1195,6 +1197,25 @@ func getProxyEnv(proxyConfig *apiconfigv1.Proxy) []corev1.EnvVar {
 	return envVars
 }
 
+func transformToolkitForCDI(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n ClusterPolicyController) {
+	if !config.CDI.IsEnabled() {
+		return
+	}
+
+	setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), CDIEnabledEnvName, "true")
+	setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), NvidiaCtrRuntimeCDIPrefixesEnvName, "nvidia.cdi.k8s.io/")
```

Comment:
Note: This is would mean that native CDI support for containers will not work when using `cdi-annotations` in the device plugin.

Reply:
OK. I see that we set this value based on whether the runtime supports CDI in the device plugin. Does that mean that we need to optionally set that here too.

Note that I think if the runtime supports CDI and it is enabled, we would want to avoid also responding to the annotations in the toolkit.

Reply:
I can update this so that we set this envvar conditionally.

On a first pass, I didn't think it was required to remove this setting when using native CDI support for containers -- since our device-plugin would be using a different annotation prefix, `cdi.k8s.io`.

Reply:
I have updated this now to conditionally set this envvar only when native CDI is not supported.